raiden/raiden_event_handler.py

```diff
@@ -237,34 +239,49 @@ def on_raiden_event(self, raiden: "RaidenService", chain_state: ChainState, even
 
     @staticmethod
     def handle_store_message(raiden: "RaidenService", store_message_event: StoreMessageEvent):
-        existing_message = LightClientMessageHandler.is_light_client_protocol_message_already_stored(
-            store_message_event.payment_id,
-            store_message_event.message_order,
-            store_message_event.message_type,
-            store_message_event.message.to_dict()["type"],
-            raiden.wal)
-        if not existing_message:
-            LightClientMessageHandler.store_light_client_protocol_message(store_message_event.message_id,
-                                                                          store_message_event.message,
-                                                                          store_message_event.is_signed,
-                                                                          store_message_event.sender_light_client_address,
-                                                                          store_message_event.receiver_light_client_address,
-                                                                          store_message_event.message_order,
-                                                                          store_message_event.message_type,
-                                                                          raiden.wal,
-                                                                          store_message_event.payment_id)
+        existing_message: LightClientProtocolMessage
+        if store_message_event.payment_id:
+            # payment related message
+            existing_message = LightClientMessageHandler.get_message_for_payment(
+                message_id=store_message_event.message_id,
+                light_client_address=store_message_event.light_client_address,
+                payment_id=store_message_event.payment_id,
+                order=store_message_event.message_order,
+                message_type=store_message_event.message_type,
+                message_protocol_type=store_message_event.message.to_dict()["type"],
+                wal=raiden.wal
+            )
         else:
-            stored_but_unsigned = existing_message.signed_message is None
-            if stored_but_unsigned and store_message_event.is_signed:
+            existing_message = LightClientMessageHandler.get_message_by_content(
+                light_client_address=store_message_event.light_client_address,
+                message_type=store_message_event.message_type,
+                message=store_message_event.message,
+                wal=raiden.wal
+            )
```

Comment:
I would extract this into a function `get_message_by_event(raiden: "RaidenService", store_message_event: StoreMessageEvent) -> Optional[LigthClientProtocolMessage]` or something like that.

Reply:
I like your suggestion, it's done, thanks